constantdict/__init__.py

```diff
@@ -170,3 +170,36 @@ def discard(self, key: K) -> constantdict[K, V]:
     setdefault = _del_attr  # type: ignore[assignment]
 
     # }}}
+
+    # {{{ mutation
+
+    def mutate(self) -> constantmutabledict[K, V]:
+        """Return a mutable version of this :class:`constantdict`."""
+        # Based on the immutables.Map API.
+        # This needs to make a copy since the original dictionary must
+        # not be modified.
+        return constantmutabledict(self)
+
+    # }}}
+
+
+class constantmutabledict(constantdict[K, V]):  # noqa: N801
+    """A mutable immutable dictionary."""
```

Comment:
How is `__getitem__` supposed to behave on this? Is it supposed to exist?

Reply:
In immutables.Map, it appears to work normally. I added a small test in [c721f51](https://github.com/matthiasdiener/constantdict/pull/20/commits/c721f51414415fe0c5b92fa957d67db7352d65c5) to demonstrate it working here too.